third_party/terraform/resources/resource_service_networking_connection.go.erb

```diff
@@ -121,11 +121,42 @@ func resourceServiceNetworkingConnectionRead(d *schema.ResourceData, meta interf
 	return nil
 }
 
-// NOTE(craigatgoogle): The API for this resource doesn't define an update, however the behavior
-// of Create serves as a de facto update by overwriting connections with the duplicate
-// tuples: (network/service).
 func resourceServiceNetworkingConnectionUpdate(d *schema.ResourceData, meta interface{}) error {
-	return resourceServiceNetworkingConnectionCreate(d, meta)
+	config := meta.(*Config)
+
+	connectionId, err := parseConnectionId(d.Id())
+	if err != nil {
+		return fmt.Errorf("Failed to find Service Networking Connection, err: %s", err)
+	}
+
+	parentService := formatParentService(connectionId.Service)
+
+	if d.HasChange("reserved_peering_ranges") {
+		network := d.Get("network").(string)
+		serviceNetworkingNetworkName, err := retrieveServiceNetworkingNetworkName(d, config, network)
+		if err != nil {
+			return fmt.Errorf("Failed to find Service Networking Connection, err: %s", err)
+		}
+
+		connection := &servicenetworking.Connection{
+			Network:               serviceNetworkingNetworkName,
+			ReservedPeeringRanges: convertStringArr(d.Get("reserved_peering_ranges").([]interface{})),
+		}
+
+		// The API docs don't specify that you can do connections/-, but that's what gcloud does,
+		// and it's easier than grabbing the connection name. The docs also mention an update mask,
```

Comment:
Given that we've seen a breaking change already, should we set the update mask to guard ourself against a future change to the API's behaviour?

Reply:
Done